### ISSUE TITLE
[rfc] Add a "core only" build flag

### ISSUE THREAD

**Issue by driazati** [pytorch/pytorch#56161]
Usually when developing on something outside of the "math" side of PyTorch (i.e. c10, utils things, jit), I end up adding a ton a build flags à la:

```
export BUILD_TEST=0
export BUILD_CAFFE2_OPS=0
export USE_NUMPY=0
export USE_KINETO=0
export USE MKLDNN=0
export USE_NNPACK=0
export USE_TENSORPIPE=0
export USE_QNNPACK=0
export USE_DISTRIBUTED=0
export USE_MPI=0
export BUILD_CAFFE2=0
```

It'd be nice if we could roll up a recommended set of these (I mostly picked these out of `setup.py`) into a single `MINIMAL_BUILD` or something that includes as few external libraries as possible

**Comment by walterddr** [comment]
Good idea. I also think we can split that into 2 categories depending on whether it purely enable/disable components in `third_party/` (such as USE_KINETO, USE MKLDNN, USE_NNPACK, USE_TENSORPIPE, USE_QNNPACK) or involve compiling part of pytorch repo (such as BUILD_CAFFE2, USE_DISTRIBUTED, BUILD_TEST)

**Comment by pearu** [comment]
Would it make sense to use https://buck.build/ to select specific pytorch components, including core-only?
AFAIK, the buck can be used to build pytorch but what is lacking is the documentation on how to actually use it for pytorch.
